### PR TITLE
Less ignored errors in Psalm

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Reflection/ReflectionParameter.php [hunk 1]
```diff
@@ -497,6 +494,31 @@ private function getClassFromNamedType(ReflectionNamedType $namedType): ?Reflect
         }
     }
 
+    private function detectIsOptional(): bool
```
Comment (Ocramius) [inline src/Reflection/ReflectionParameter.php]: This seems to be improved here, but not really tested 🤔

**Review by Ocramius — APPROVED** [review]
Nice cleanup!

File: src/Reflection/ReflectionParameter.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionParameter.php]: Rebased :)

Comment (Ocramius) [reply src/Reflection/ReflectionParameter.php]: Indeed, solved in #847

**Comment by Ocramius** [comment]
I did a skim of the escaped mutations, and **a lot** of them should probably really be caught by static analysis: weird :-\ 

It may be that our `psalm.xml` is still too lenient, but we'll improve it over time.

**Comment by Ocramius** [comment]
Thanks @kukulich!

**Comment by Ocramius** [comment]
Ok, indeed, a lot of the mutations that are left seem to be covered by nullability scenarios, specifically part of our remaining `psalm.xml` suppressions:

```xml
    <issueHandlers>
        <NullableReturnStatement errorLevel="suppress" />
        <InvalidNullableReturnType errorLevel="suppress" />

        <MoreSpecificReturnType errorLevel="suppress" />
        <LessSpecificReturnStatement errorLevel="suppress" />

        <PossiblyNullArgument errorLevel="suppress" />
    </issueHandlers>
```

Getting rid of most of them should get us a much better mutation testing profile too 💪

**Pull request merged**